src/jsMain/kotlin/ui/configuration/PhonemesConversion.kt

```diff
@@ -138,12 +138,17 @@ private fun ChildrenBuilder.buildPhonemesConversionDetail(
                                         )
                                     }
                                 }
-                                PhonemesMappingRequest.Presets.forEach { preset ->
-                                    MenuItem {
-                                        value = preset.first
-                                        +(preset.first)
+                                PhonemesMappingRequest.Presets
+                                    .filter { preset ->
+                                        state.sourceFormat in preset.sourceFormats &&
```

Comment:
Instead of adding these fields to the State, could you please directly use the values from the `props`? These are not changed so they don't have to be included in the state.